storage/Filesystems/BlkFilesystemImpl.cc

```diff
@@ -708,43 +728,40 @@ namespace storage
     Text
     BlkFilesystem::Impl::do_create_text(Tense tense) const
     {
-	// TODO handle multiple BlkDevices
-
-	const BlkDevice* blk_device = get_blk_device();
-
 	Text text = tenser(tense,
 			   // TRANSLATORS: displayed before action,
 			   // %1$s is replaced by file system name (e.g. ext4),
-			   // %2$s is replaced by device name (e.g. /dev/sda1),
-			   // %3$s is replaced by size (e.g. 2GiB)
-			   _("Create %1$s on %2$s (%3$s)"),
+			   // %2$s is replaced by one or more devices (e.g /dev/sda1 (1.0 GiB) and
+			   // /dev/sdb2 (1.0 GiB))
+			   _("Create %1$s on %2$s"),
 			   // TRANSLATORS: displayed during action,
 			   // %1$s is replaced by file system name (e.g. ext4),
-			   // %2$s is replaced by device name (e.g. /dev/sda1),
-			   // %3$s is replaced by size (e.g. 2GiB)
-			   _("Creating %1$s on %2$s (%3$s)"));
+			   // %2$s is replaced by one or more devices (e.g /dev/sda1 (1.0 GiB) and
+			   // /dev/sdb2 (1.0 GiB))
+			   _("Creating %1$s on %2$s"));
 
-	return sformat(text, get_displayname(), blk_device->get_name(),
-		       blk_device->get_impl().get_size_text());
+	return sformat(text, get_displayname(), join(get_blk_devices(), JoinMode::COMMA, 10));
     }
 
 
     Text
     BlkFilesystem::Impl::do_set_label_text(Tense tense) const
     {
-	const BlkDevice* blk_device = get_blk_device();
-
 	Text text = tenser(tense,
 			   // TRANSLATORS: displayed before action,
-			   // %1$s is replaced by device name (e.g. /dev/sda1),
-			   // %2$s is replaced by label (e.g. ROOT)
-			   _("Set label of %1$s to %2$s"),
+			   // %1$s is replaced by file system name (e.g. ext4),
+			   // %2$s is replaced by one or more devices (e.g /dev/sda1 (1.0 GiB) and
+			   // /dev/sdb2 (1.0 GiB)),
+			   // %3$s is replaced by label (e.g. ROOT)
+			   _("Set label of %1$s on %2$s to %3$s"),
 			   // TRANSLATORS: displayed during action,
-			   // %1$s is replaced by device name (e.g. /dev/sda1),
-			   // %2$s is replaced by label (e.g. ROOT)
-			   _("Setting label of %1$s to %2$s"));
+			   // %1$s is replaced by file system name (e.g. ext4),
+			   // %2$s is replaced by one or more devices (e.g /dev/sda1 (1.0 GiB) and
+			   // /dev/sdb2 (1.0 GiB)),
+			   // %3$s is replaced by label (e.g. ROOT)
+			   _("Setting label of %1$s on %2$s to %3$s"));
 
-	return sformat(text, blk_device->get_name(), label);
+	return sformat(text, get_displayname(), join(get_blk_devices(), JoinMode::COMMA, 10), label);
```

Comment:
Since this `join(...)` is used so many times, maybe add a function for it? 

```C++
string get_blk_devices_list( join_mode, int max_count );
```

?

Reply:
That does not look really shorter to me.